catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/data/impl/BasicTypes.java

```diff
@@ -99,6 +101,8 @@
 
     public static final String VALIDATION_ERRORS = "validation-errors";
 
+    private static final Map<String, AttributeType> ATTRIBUTE_TYPE_MAP = new HashMap<>();
+
```

Comment:
We can, but my point was that we can't assign values to the `public static final` `AttributeType`s in a separate method because they are `final`. We could do something like this:
```
static {
    BOOLEAN_TYPE = addAttributeType("BOOLEAN_TYPE", AttributeFormat.BOOLEAN, Boolean.class);
    DOUBLE_TYPE = addAttributeType(...);
    ...
}

private static <T extends Serializable> AttributeType<T> addAttributeType(String typeName, AttributeFormat attributeFormat, Class<T> bindingClass) {
    AttributeType<T> attributeType = new ConcreteAttributeType<>(attributeFormat, bindingClass);
    ATTRIBUTE_TYPE_MAP.put(typeName, attributeType);
    return attributeType;
}
```